extensions/oidc-db-token-state-manager/runtime/src/main/java/io/quarkus/oidc/db/token/state/manager/runtime/OidcDbTokenStateManagerInitializer.java

```diff
@@ -0,0 +1,145 @@
+package io.quarkus.oidc.db.token.state.manager.runtime;
+
+import static io.quarkus.oidc.db.token.state.manager.runtime.OidcDbTokenStateManager.now;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import jakarta.enterprise.event.Observes;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.Pool;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+
+public class OidcDbTokenStateManagerInitializer {
+
+    private static final Logger LOG = Logger.getLogger(OidcDbTokenStateManagerInitializer.class);
+    private static final String FAILED_TO_CREATE_DB_TABLE = "unknown reason, please report the issue and create table manually";
+    /**
+     * Extra 30 seconds before we delete expired tokens.
+     */
+    private static final long EXPIRED_EXTRA_GRACE = 30;
+
+    void initialize(@Observes StartupEvent event, OidcDbTokenStateManagerRunTimeConfig config, Vertx vertx, Pool pool,
+            OidcDbTokenStateManagerInitializerProperties initializerProps) {
+        if (config.createDatabaseTableIfNotExists()) {
+            createDatabaseTable(pool, initializerProps.createTableDdl, initializerProps.supportsIfTableNotExists);
+        }
+        periodicallyDeleteExpiredTokens(vertx, pool, config.deleteExpiredDelay().toMillis());
+    }
+
+    private static void periodicallyDeleteExpiredTokens(Vertx vertx, Pool pool, long delayBetweenChecks) {
+        vertx
+                .setPeriodic(5000, delayBetweenChecks, new Handler<Long>() {
```

Comment:
Should the first and second parameters be equal ?

Reply:
the first parameter says before first periodic is started, wait for 5 seconds. Logic in there is that if we do many things right before app started, it can be a lot <= that's very vague intentionally, we can't know what are all the things that user will do on app start (or what extension is using), giving some times decrease overload in my mind.

> Should the first and second parameters be equal ?

It would mean that first delete is after 8 hours. What if app restarted in the middle of interval? It would mean 4 h + 8 h. But yeah, I see were you are heading to. No idea, do you prefer it to be same?

Reply:
@michalvavrik I see, sorry